### PR TITLE
Travis CI: Use pyenv on macOS

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/pyvec/naucse_render/issues/17

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,27 +16,26 @@ matrix:
...
-      env:
-        - PYTHON=python3
-        - HOMEBREW_NO_INSTALL_CLEANUP=1
       before_install:
-        # Remove unneeded stuff (which Brew would want to upgrade)
-        - "brew remove numpy gdal postgis postgresql || :"
-        # Install or update Python
-        - "brew install python3 || :"
-        - "brew upgrade python3"
+        - HOMEBREW_NO_INSTALL_CLEANUP=1 brew upgrade pyenv
```
Comment (hroncok) [inline .travis.yml]: this is optional, currently gives 3.7.6 over 3.7.5

**Comment by encukou** [comment]
It looks like this will technically work, but it'll build CPython from source for every pull request. I don't want to do that to poor Travis.

I've added this to https://github.com/pyvec/naucse_render/pull/18 to see what happens.
I let Travis build it once and then restarted the job. Here's the output of the rebuild: https://travis-ci.org/pyvec/naucse_render/jobs/634235527#. The installation takes 3+ minutes.

**Comment by hroncok** [comment]
is 3+ minutes good or bad?

**Comment by encukou** [comment]
Not too bad for compiling CPython! But I don't think compiling CPython shouldn't be part of the CI check here.

**Comment by hroncok** [comment]
Too bad the cache does not contain the compiled interpreter :(

**Comment by encukou** [comment]
What's in the cache, anyway?

**Comment by hroncok** [comment]
I guess it is the downloaded source code. But no idea really.

**Comment by encukou** [comment]
We're off Travis now.

**Pull request closed without merging**